docs/source/extending/ui.rst

```diff
@@ -0,0 +1,67 @@
+User Interface
+==============
+
+UI Hooks
+--------
+
+UI Hooks are callbacks that are triggered at some events during the lifetime
+of the UI. It has no model. The factory must inherit from
+``BaseUIHooksFactory``, and must reimplement ``get_ui_hooks_manager_class()``
+to return a class inheriting from ``BaseUIHooksManager``. This class has
+specific methods to be reimplemented to perform operations before and after
+some UI operations.
+
+Any ``BaseDriverEvents`` that are required to be delivered to a UI can be indicated
+using the ``UIEventMixin`` class. The ``MCOStartEvent``, ``MCOProgressEvent`` and
+``MCOFinishEvent`` are all examples of such objects.
```

Comment:
This section is probably out of scope for our tutorial at the moment - we can always put it in later / point to it  in the BDSS repo docs if needs be.